src/Mutation/MutationGenerator.php

```diff
@@ -118,9 +118,7 @@ public function generate(bool $onlyCovered, array $extraNodeVisitors = []): arra
         $this->eventDispatcher->dispatch(new MutationGeneratingStarted(count($this->sourceFiles)));
 
         foreach ($this->sourceFiles as $file) {
-            if (!$onlyCovered || $this->hasTests($file)) {
```

Comment:
I moved this check in the `getMutationsFromFile()` which will be more in line with what's coming next